src/app/features/auth/components/reset-password/reset-password.component.ts

```diff
@@ -12,32 +17,54 @@ import { InputFieldComponent } from '../input-field/input-field.component';
   templateUrl: './reset-password.component.html',
   styleUrl: './reset-password.component.sass'
 })
-export class ResetPasswordComponent {
+export class ResetPasswordComponent implements OnDestroy{
   passwordForm: FormGroup = this.fb.group({
     password: ['', [Validators.required, Validators.minLength(12), passwordValidator]],
     confirmPassword: ['', Validators.required]
   }, {
-    validators: this.matchPassword
+    validators: passwordMatchValidator()
   });
 
+  private subscription: Subscription = new Subscription(); 
   
-  constructor(private fb: FormBuilder) {}
-  matchPassword(group: FormGroup) {
-    const password = group.get('password')?.value;
-    const confirmPassword = group.get('confirmPassword')?.value;
-
-    if (password !== confirmPassword) {
-      return { mismatchedPassword: true };
-    }
-    return null;
-  }
+  constructor(private fb: FormBuilder,private router:Router,private resetPasswordService:ResetPasswordService,@Inject(NOTYF) private notyf: Notyf,   private route: ActivatedRoute,) {}
 
 
 
   onSubmit() {
+
     if (this.passwordForm.valid) {
+      const {password,confirmPassword} = this.passwordForm.value;
+    
+      this.route.queryParamMap.subscribe((params) => {
+        const token = params.get('token'); 
+        if(token){
+        this.resetPasswordService.resetPassword(password,confirmPassword,token).subscribe({
+          next: (response) => {
+            console.log('Response:', response)
+            this.notyf.success('Password Reset Successfully');
+            this.passwordForm.reset();
+            this.router.navigateByUrl('/login');
+          },
   
-    } 
+          error: (error) => {
+            console.log('Error:', error);
+            this.notyf.error('Password Reset Failed. Please Try Again');
+          }
+        })
```

Comment:
If there is a network error we can't show users the message in the notyf.error 
That will be confusing to users okay